packages/hash/api/src/db/adapter.ts

```diff
@@ -72,8 +88,36 @@ export type DbLink = {
   removedFromSourceAt?: Date;
   removedFromSourceByAccountId?: string;
 
+  /**
+   * Question: When would we intend that `destinationAccountId` is different than `sourceAccountId`
+   *  * Does this have negative implications for sharding?
+   *  * e.g. When all items from Account A point to many items in Account B
+   */
   destinationAccountId: string;
+
+  /**
+   * Idea 1/10:
+   * Consider making the target of the link actually a composite of:
+   *    entity_type_id + key_property_values e.g. `{ entityType: "GithubIssue", where: { id: 1236728 } }`
+   *  * This could allow us to better support out of order ingestion.
+   *  * This could implicate that we need a view or something to join in pgSQL instead of us proactively
+   *    updating custom link index tables. (e.g. for canonicalization)
+   */
   destinationEntityId: string;
+
+  /**
+   * Optional way to pin the link to a specific
+   * version of the destination entity.
+   *
+   * Progress 0/10:
+   *
+   * Maybe a smart idea if the way we consider internal
+   * versions as something mapping to versions of an use-cases
+   * domain model like specific dependency version.
+   * But, at that point would it make sense to instead store
+   * separate entities for each "dependency version" or
+   * "license version", etc?
+   */
```

Comment:
Interesting idea, I think that's definitely worth considering so have created a task: https://app.asana.com/0/1200211978612931/1202085856561985/f

Don't think the idea should live as a comment here though, as we currently only have todos and docs in our codebase:
```suggestion
  /**
   * Optional way to pin the link to a specific
   * version of the destination entity.
   */
```